kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -37,13 +39,44 @@ grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
     ctrl.selectedCollections = selectedCollections;
 }]);
 
+grCollectionsPanel.factory('collectionsTreeState', ['$window', function($window) {
+    // TODO: Add garbage collection to state.
+    const localStorageKey = 'collectionsTreeOpen';
+    const jsonStr = $window.localStorage.getItem(localStorageKey) || '[]';
+    const jsonArr = JSON.parse(jsonStr);
+    // A little bit of security incase this was set weirdly before.
+    // The var is use so we don't have to read from localStorage the whole time.
+    var stateCache = Array.isArray(jsonArr) ? new Set(jsonArr) : new Set();
+
+
+    function setState(pathId, show) {
+        const newState = show ? stateCache.add(pathId) : stateCache.delete(pathId);
```

Comment:
AFAICS you're using ES6 Sets, not Immutable Sets, so `add` and `delete` would mutate `stateCache` too?